### PR TITLE
[ADD] Adds account_outstanding_payment

### PR DESCRIPTION
**Pull request by daramousk**
module. Backporting functionality from version 9. Allow an invoice to be paid using already existing customer credit

### CODE REVIEW

File: account_outstanding_payments/README.rst [hunk 1]
```diff
@@ -0,0 +1,66 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+    :target: http://www.gnu.org/licenses/agpl-3.0-standalone.html
+    :alt: License: AGPL-3
+
+============================
+account_outstanding_payments
```
Comment (hbrunn) [inline account_outstanding_payments/README.rst]: use some nice human readable string

File: account_outstanding_payments/README.rst [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+=====
+
+To use this module, you need to:
+
+#. go to ...
+.. image:: https://odoo-community.org/website/image/ir.attachment/5784_f2813bd/datas
+    :alt: Try me on Runbot
+    :target: https://runbot.odoo-community.org/runbot/{repo_id}/8.0
+
+.. repo_id is available in https://github.com/OCA/maintainer-tools/blob/master/tools/repos_with_ids.txt
```
Comment (hbrunn) [inline account_outstanding_payments/README.rst]: expand the variable above using this info

File: account_outstanding_payments/README.rst [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+Known issues / Roadmap
+======================
+
+* ...
```
Comment (hbrunn) [inline account_outstanding_payments/README.rst]: delete empty sections

File: account_outstanding_payments/__openerp__.py [hunk 4]
```diff
@@ -0,0 +1,27 @@
+# -*- coding: utf-8 -*-
+# © 2017 Therp BV <http://therp.nl>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+{
+    "name": "Account Outstanding Payments",
+    "version": "8.0.1.0.0",
+    "author": "Therp BV,Odoo Community Association (OCA)",
+    "license": "AGPL-3",
+    "category": "",
+    "summary": "",
```
Comment (hbrunn) [inline account_outstanding_payments/__openerp__.py]: those two fields are mandatory

File: account_outstanding_payments/__openerp__.py [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+        'views/templates.xml',
+        'views/account_invoice_view.xml',
+        'views/account_config_settings_view.xml',
+        'security/ir.model.access.csv',
+    ],
+    "qweb": [
+        'static/src/xml/account_payment.xml',
+    ], 
+    "test": [
+    ],
```
Comment (hbrunn) [inline account_outstanding_payments/__openerp__.py]: delete empty keys

File: account_outstanding_payments/models/account_invoice.py [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+from openerp.tools.translate import _
+import json
+from openerp.tools.float_utils import float_is_zero
+
+
+class AccountInvoice(models.Model):
+    _inherit = 'account.invoice'
+
+    @api.one
+    def _get_outstanding_info_JSON(self):
```
Comment (hbrunn) [inline account_outstanding_payments/models/account_invoice.py]: make clear where you borrowed code from 9

File: account_outstanding_payments/security/ir.model.access.csv [hunk 7]
```diff
@@ -0,0 +1 @@
+"id","name","model_id:id","group_id:id","perm_read","perm_write","perm_create","perm_unlink"
```
Comment (hbrunn) [inline account_outstanding_payments/security/ir.model.access.csv]: delete empty files

**Review by hbrunn — CHANGES_REQUESTED** [review]
you really need to attribute the code you took from 9 to the respective authors. Either do this on the git level by just cherry picking/filtering the commits involved in the pieces of code, or add comments in the code linking to where the original lives. And then given that the code is tainted license wise, do as https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#backporting-odoo-modules says

**Comment by daramousk** [comment]
@hbrunn Implemented

**Comment by daramousk** [comment]
@hbrunn The original authors have been attributed on the README and on the functions that where used. Runbot fails but on a different module than this.

**Comment by lfreeke** [comment]
@hbrunn ping

**Review by hbrunn — APPROVED** [review]

**Comment by pedrobaeza** [comment]
Please use singular in module name: https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#modules

**Comment by daramousk** [comment]
@pedrobaeza Renamed
Edit: Ignore, will rename properly in a few minutes

**Comment by daramousk** [comment]
@pedrobaeza Ping

**Comment by daramousk** [comment]
@pedrobaeza Anything else to be done on this? The runbot failure I am not sure if it is coming from here.

**Comment by pedrobaeza** [comment]
Well, you should find reviewers for your work. You can review other PRs and ask in exchange to review yours.

**Comment by daramousk** [comment]
@NL66278 Can you take a look at this?

**Comment by moylop260** [comment]
> ...Anything else to be done on this? The runbot failure I am not sure if it is coming from here.

@daramousk 
The log of runbot 

 - screen shot 2017-11-16 at 09 19 36

Show a yellow status because a warning log:

https://runbot2.odoo-community.org/runbot/static/build/3309651-302-8b43e4/logs/job_20_test_all.txt

```bash
2017-11-09 12:48:29,840 160 WARNING openerp_test openerp.modules.loading: The model account.invoice.stage has no access rules, consider adding one. E.g. access_account_invoice_stage,access_account_invoice_stage,model_account_invoice_stage,,1,0,0,0
```

You are creating a new model but has no access rules.

**Comment by daramousk** [comment]
@moylop260 Well, this model is not created from this module.

**Review by moylop260 — APPROVED** [review]

**Comment by daramousk** [comment]
@hbrunn Possibly this is ready to be merged?

**Comment by hbrunn** [comment]
weird, this was green when I merged. @daramousk please check https://travis-ci.org/OCA/account-invoicing/jobs/432831552#L1500 and submit a followup to fix the test

**Pull request merged**